### PR TITLE
ci: migrating to GitHub actions

### PR DESCRIPTION
**Pull request by JBAhire**
## Description
Migrating to GitHub actions from CircleCI as per https://github.com/hypertrace/hypertrace/issues/144

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
- https://docs.github.com/en/free-pro-team@latest/actions

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+on:
+# Will only run when release is published. 
+  pull_request:
+  release:
+    types: 
+      - created
+  workflow_dispatch:
+
+jobs:
+  validate-helm-charts:
```
Comment (kotharironak) [inline .github/workflows/publish.yml:11]: Isn't this requires any building task? Anyone can commit to this repo w/o build?

Comment (kotharironak) [reply .github/workflows/publish.yml:11]: I see, it has only helm charts in this repo.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
```
Comment (kotharironak) [inline .github/workflows/publish.yml:17]: Don't need this in the context we are running,

```
ref: ${{github.event.pull_request.head.ref}}
          repository: ${{github.event.pull_request.head.repo.full_name}}
```

**Review by kotharironak — DISMISSED** [review]

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+name: Publish artifacts
```
Comment (aaron-steinfeld) [inline .github/workflows/publish.yml:1]: These are unrelated jobs with different triggers, aren't they? Can we separate them into two different workflows?

Comment (JBAhire) [reply .github/workflows/publish.yml:1]: we can but as this repo only has charts so thought of combining them and using if condition as running validate charts on release is fine anyways.

Comment (JBAhire) [reply .github/workflows/publish.yml:1]: Should I separate them?

Comment (aaron-steinfeld) [reply .github/workflows/publish.yml:1]: I think so - there's very little duplication - just the triggers themselves. We also want to make sure we run validate on push (which should make it unnecessary on release). It also saves us from having the publish job show up as skipped on every PR and allows us to keep this naming more consistent: `Publish artifacts / validate-helm-charts ` -> `Pull Request/ validate-helm-charts `

Comment (JBAhire) [reply .github/workflows/publish.yml:1]: cool

Comment (JBAhire) [reply .github/workflows/publish.yml:1]: addressed

**Review by aaron-steinfeld — APPROVED** [review]

**Review by buchi-busireddy — APPROVED** [review]
